luatest/server.lua

```diff
@@ -547,24 +538,44 @@ function Server:exec(fn, args, options)
         error('net_box is not connected', 2)
     end
 
-    local ups = utils.upvalues(fn)
-    if next(ups) ~= nil then
+    local autorequired_pkgs = {'luatest'}
+    local autorequired_ups = {}
+    local other_ups = {}
+    for i = 1, debug.getinfo(fn, 'u').nups do
+        local is_other_up = true
+        local name, value = debug.getupvalue(fn, i)
+        for _, p in ipairs(autorequired_pkgs) do
+            if package.loaded[p] == value then
+                autorequired_ups[i] = p
+                is_other_up = false
+                break
+            end
+        end
+        if is_other_up then
```

Comment:
Nit: It can be chacked as `autorequired_ups[i] ~= nil` without extra `is_other_up` variable. (Or as `autorequired_ups[name] ~= nil` if you'll accept my comment regarding upvalues matching by names.)

Reply:
Fixed.